### DISCUSSION TITLE
Is there a way to generate a path URL to pass into creating a window with Tauri?

### DISCUSSION THREAD

**Discussion by zaytri** [TanStack/router#1503] [Q&A]
I'm trying to figure out how to nicely combine TanStack Router with [Tauri](https://tauri.app/) (desktop app framework similar to Electron). Out of the box it works perfectly for routes in a single window. I am also able to create windows that load specific routes, by manually passing the URL of those routes. Is there some kind of function I could use so that it could build the URL for me, similar to how it gets built in the `Link` component? This way I could create my own component that benefits from TypeScript to open links that create windows.

**Comment by SeanCassiere** [comment]
Building a href can be done using the `buildLocation` method on the Router instance. This method returns an object with a `href` property.

**Reply by zhouzi** [reply]
I could be wrong but from what I understand that's not type-safe, correct?

Here's en example:

```ts
router.buildLocation({
  to: Route.to,
  params: {
    // This is not type-checked
    foo: "foo",
  }
});
```

**Reply by schiller-manuel** [reply]
@zhouzi it is typesafe. please check that you are using a recent version as there was a regression that has been fixed

**Reply by zhouzi** [reply]
Oh yeah, right, my bad. It is typesafe indeed 💯
